### PR TITLE
Documentation: Add documentation for building coreclr for emulator

### PR DESCRIPTION
**Pull request by prajwal-aithal**
Adding documentation for building coreclr for the Linux ARM Emulator.

Signed-off-by: Prajwal A N <an.prajwal@samsung.com>

### CODE REVIEW

**Comment by prajwal-aithal** [comment]
@leemgs @benpye Could you please review this?

This PR can be merged right after #6050 is merged (as the instructions given in the updated documentation are based on the changes made in #6050 ).

File: Documentation/building/cross-building.md [hunk 1]
```diff
@@ -95,4 +95,40 @@ The output is at bin/Product/<BuildOS>.arm.Debug/mscorlib.dll.
     (console) ARMv7 Thumb Mono/.Net assembly, for MS Windows
 ```
 
+Building coreclr for Linux ARM Emulator
+=======================================
 
+It is possible to build coreclr binaries (native and mscorlib.dll) and run coreclr unit tests on the Linux ARM Emulator (latest version provided here: [#3805](https://github.com/dotnet/coreclr/issues/3805)).
+The `tests/scripts/arm32_ci_script.sh` script does this.
```
Comment (leemgs) [inline Documentation/building/cross-building.md]: My comment is same such as #10029.

File: Documentation/building/cross-building.md [hunk 2]
```diff
@@ -95,4 +95,40 @@ The output is at bin/Product/<BuildOS>.arm.Debug/mscorlib.dll.
...
+* Build corefx binaries for the Emulator as given [here](https://github.com/dotnet/corefx/blob/master/Documentation/building/cross-building.md#building-corefx-for-linux-arm-emulator).  
+Setup these binaries at `~/corefx/bin/Linux.AnyCPU.Release`, `~/corefx/bin/Unix.AnyCPU.Release`, and `~/corefx/bin/AnyOS.AnyCPU.Release` (accordingly for Debug configuration).
+* Run the following command (change value of `--testDirFile` argument to the file containing your selection of tests):
+```
+prajwal@ubuntu ~/coreclr $ ./tests/scripts/arm32_ci_script.sh \
+    --emulatorPath=/opt/linux-arm-emulator \
+    --mountPath=/opt/linux-arm-emulator-root \
+    --buildConfig=Release \
+    --testRootDir=~/coreclr/bin/Windows_NT.x64.Release \
+    --coreFxNativeBinDir=~/corefx/bin/Linux.arm-softfp.Release \
```
Comment (leemgs) [inline Documentation/building/cross-building.md]: Could you describe the difference between 'arm' and 'arm-softfp' in more detail?

**Comment by leemgs** [comment]
\CC: @sjsinju, FYI

**Comment by prajwal-aithal** [comment]
@leemgs Can you review the updated PR?

**Comment by leemgs** [comment]
It seems good to me. :100: 
Acked-by: Geunsik Lim <geunsik.lim@samsung.com>

File: Documentation/building/cross-building.md [hunk 3]
```diff
@@ -95,4 +95,42 @@ The output is at bin/Product/<BuildOS>.arm.Debug/mscorlib.dll.
...
+Setup the binaries at `~/coreclr/bin/tests/Windows_NT.x64.Release`.
+* Build corefx binaries for the Emulator as given [here](https://github.com/dotnet/corefx/blob/master/Documentation/building/cross-building.md#building-corefx-for-linux-arm-emulator).  
+Setup these binaries at `~/corefx/bin/Linux.arm-softfp.Release`, `~/corefx/bin/Linux.AnyCPU.Release`, `~/corefx/bin/Unix.AnyCPU.Release`, and `~/corefx/bin/AnyOS.AnyCPU.Release` (accordingly for Debug configuration).
+* Run the following command (change value of `--testDirFile` argument to the file containing your selection of tests):
+```
+prajwal@ubuntu ~/coreclr $ ./tests/scripts/arm32_ci_script.sh \
+    --emulatorPath=/opt/linux-arm-emulator \
+    --mountPath=/opt/linux-arm-emulator-root \
+    --buildConfig=Release \
+    --testRootDir=~/coreclr/bin/Windows_NT.x64.Release \
```
Comment (sjsinju) [inline Documentation/building/cross-building.md]: In 123 line, you described "+Setup the binaries at `~/coreclr/bin/tests/Windows_NT.x64.Release`.". But this example missed the 'test ' folder name between '/bin/' and '/Windows_NT.x64.Release'. So If user just copy and paste, It seems to make some problems.

**Comment by prajwal-aithal** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build

**Comment by prajwal-aithal** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build

**Comment by prajwal-aithal** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build

**Comment by prajwal-aithal** [comment]
@mmitche @jashook Can we merge this PR? This PR adds documentation for the newly enabled method of cross building and testing for the Linux ARM Emulator.

**Comment by prajwal-aithal** [comment]
The Linux ARM Emulator Release Build failure can be ignored for now. @sjsinju and @jyoungyun are working on it.

**Comment by sjsinju** [comment]
LGTM

**Comment by mmitche** [comment]
LGTM

**Comment by jashook** [comment]
lgtm

**Pull request merged**